src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -344,6 +348,40 @@ protected final void handleSelectedKey(final SelectionKey sk) throws IOException
 
   }
 
+  private void handleFullTasksCircularBuffer(Runnable r, int sleepMillis)
+  {
+    final Thread currentThread = Thread.currentThread();
+    if (eventThread == currentThread) {
+      int size = tasks.size();
+      if (size == tasks.capacity()) {
+        try {
+          while (size > 0) {
+            tasks.pollUnsafe().run();
+            size--;
+          }
+        } catch (RuntimeException e) {
+          logger.error("", e);
+          throw e;
+        } catch (Exception e) {
+          logger.error("", e);
```

Comment:
then who is the consumer of the runtime exception which is being raised immediately after this line?